docs/06-concepts/05-sessions.md

```diff
@@ -1,26 +1,476 @@
+
+```dart
+Stream<Message> chat(Session session, Stream<String> incoming) async* {
+  // MethodStreamSession stays open for the stream duration
+  await for (var message in incoming) {
+    yield Message(text: message, time: DateTime.now());
+  }
+  // Closes when stream completes
+}
+```
+
+### FutureCallSession - Background tasks
+
+Created for: Scheduled operations
+Lifetime: Task execution duration
+Common in: Email sending, report generation, data cleanup
+
+```dart
+class EmailSender extends FutureCall {
+  @override
+  Future<void> invoke(Session session, EmailData? data) async {
+    // FutureCallSession created just for this task
+    await sendEmail(session, data!);
+    // Closes when task completes
+  }
+}
+```
+
+### InternalSession - System operations
+
+Created for: Manual background work
+Lifetime: Until explicitly closed
+Common in: Database migrations, batch imports, maintenance tasks
+
+```dart
+// Cleanup old data periodically
+Future<void> cleanupOldData() async {
+  var session = await Serverpod.instance.createSession(
+    enableLogging: false, // Often disabled for system tasks
+  );
+  try {
+    await OldData.db.deleteWhere(
+      session,
+      where: (t) => t.createdAt < DateTime.now().subtract(Duration(days: 90)),
+    );
+  } finally {
+    await session.close(); // Must close manually
+  }
+}
+```
+
+## Best practices
+
+### 1. Let Serverpod manage sessions when possible
+
+Prefer using the session provided to your endpoint rather than creating new ones:
+
+```dart
+// ✅ Good - Use provided session
+Future<List<User>> getActiveUsers(Session session) async {
+  return await User.db.find(
+    session,
+    where: (t) => t.isActive.equals(true),
+  );
+}
+
+// ❌ Avoid - Creating unnecessary session
+Future<List<User>> getActiveUsers(Session session) async {
+  var newSession = await Serverpod.instance.createSession();
+  try {
+    return await User.db.find(newSession, ...);
+  } finally {
+    await newSession.close();
+  }
+}
+```
+
+### 2. Use FutureCalls for delayed operations
+
+Instead of managing sessions for async work, use Serverpod's future call system:
+
+```dart
+// ✅ Good - Let Serverpod manage the session
+await serverpod.futureCallWithDelay(
+  'processPayment',
+  PaymentData(orderId: order.id),
+  Duration(hours: 1),
+);
+
+// ❌ Complex - Manual session management
+Future.delayed(Duration(hours: 1), () async {
+  var session = await Serverpod.instance.createSession();
+  try {
+    await processPayment(session, order.id);
+  } finally {
+    await session.close();
+  }
+});
+```
+
+### 3. Register cleanup with will-close listeners
+
+For custom resources that need cleanup:
+
+```dart
+Future<void> processWithExternalService(Session session) async {
+  var service = await ExternalService.connect();
+
+  // Ensure cleanup even if session closes unexpectedly
+  session.addWillCloseListener((session) async {
+    await service.disconnect();
+  });
+
+  // Use service...
+}
+```
```

Comment:
Suggestion: This should be described in a section around the lifecycle rather than under best practices as this is a core concept not a style preference.